test/e2e/test/logstash/builder.go

```diff
@@ -116,6 +117,38 @@ func (b Builder) WithServices(services ...logstashv1alpha1.LogstashService) Buil
 	return b
 }
 
+func (b Builder) WithPipelines(pipelines []commonv1.Config) Builder {
+	b.Logstash.Spec.Pipelines = pipelines
+	return b
+}
+
+func (b Builder) WithPipelinesConfigRef(ref commonv1.ConfigSource) Builder {
+	b.Logstash.Spec.PipelinesRef = &ref
+	return b
+}
+
+func (b Builder) WithVolumes(vols ...corev1.Volume) Builder {
+	b.Logstash.Spec.PodTemplate.Spec.Volumes = append(b.Logstash.Spec.PodTemplate.Spec.Volumes, vols...)
+	return b
+}
+
+func (b Builder) WithVolumeMounts(mounts ...corev1.VolumeMount) Builder {
+	if b.Logstash.Spec.PodTemplate.Spec.Containers == nil {
+		b.Logstash.Spec.PodTemplate.Spec.Containers = []corev1.Container{
+			{
+				Name:         "logstash",
+				VolumeMounts: mounts,
+			},
+		}
+	} else {
+		if b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts == nil {
+			b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{}
+		}
+		b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts = append(b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts, mounts...)
+	}
+	return b
```

Comment:
nit: reduce nesting with early return
```suggestion
	if b.Logstash.Spec.PodTemplate.Spec.Containers == nil {
		b.Logstash.Spec.PodTemplate.Spec.Containers = []corev1.Container{
			{
				Name:         "logstash",
				VolumeMounts: mounts,
			},
		}
		return b
	}

	if b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts == nil {
		b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{}
	}
	b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts = append(b.Logstash.Spec.PodTemplate.Spec.Containers[0].VolumeMounts, mounts...)

	return b
```